addon/background.js

```diff
@@ -1,41 +1,41 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this
- * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-/* global browser */
-
-function log(...args) {
-  // console.log(...["[TESTPILOT v2] (background)"].concat(args)); // eslint-disable-lint no-console
-}
-
-const storage = browser.storage.local;
-const RESOURCE_UPDATE_INTERVAL = 4 * 60 * 60 * 1000; // 4 hours
-const ONE_DAY = 60 * 60 * 1000 * 24;
-const TWO_WEEKS = 2 * 7 * ONE_DAY;
-
-/* browser action constants */
-const BROWSER_ACTION_LINK_BASE = [
-  "/experiments",
-  "?utm_source=testpilot-addon",
-  "&utm_medium=firefox-browser",
-  "&utm_campaign=testpilot-doorhanger"
-].join("");
-const BROWSER_ACTION_LINK_NOT_BADGED = BROWSER_ACTION_LINK_BASE +
-      "&utm_content=not+badged";
-const BROWSER_ACTION_LINK_BADGED = BROWSER_ACTION_LINK_BASE +
-      "&utm_content=badged";
-let BROWSER_ACTION_LINK = BROWSER_ACTION_LINK_NOT_BADGED;
-
-const resources = {
-  experiments: [],
-  news_updates: []
-};
-
-let currentEnvironment = {
-  name: "production",
-  baseUrl: "https://testpilot.firefox.com"
-};
-
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/* global browser */
+
+function log(...args) {
+  // console.log(...["[TESTPILOT v2] (background)"].concat(args)); // eslint-disable-lint no-console
+}
+
+const storage = browser.storage.local;
+const RESOURCE_UPDATE_INTERVAL = 4 * 60 * 60 * 1000; // 4 hours
+const ONE_DAY = 60 * 60 * 1000 * 24;
+const TWO_WEEKS = 2 * 7 * ONE_DAY;
+
+/* browser action constants */
+const BROWSER_ACTION_LINK_BASE = [
+  "/experiments",
+  "?utm_source=testpilot-addon",
+  "&utm_medium=firefox-browser",
+  "&utm_campaign=testpilot-doorhanger"
+].join("");
+const BROWSER_ACTION_LINK_NOT_BADGED = BROWSER_ACTION_LINK_BASE +
+      "&utm_content=not+badged";
+const BROWSER_ACTION_LINK_BADGED = BROWSER_ACTION_LINK_BASE +
+      "&utm_content=badged";
+let BROWSER_ACTION_LINK = BROWSER_ACTION_LINK_NOT_BADGED;
+
+const resources = {
+  experiments: [],
+  news_updates: []
+};
+
+let currentEnvironment = {
```

Comment:
Rather than do a search/replace on all of `background.js` to replace these two strings, it might be better to create a separate new `currentEnvironment.js` file. 

Just something with contents like this, written out by `process-with-env`:
```javascript
let currentEnvironment = {
  name: "production",
  baseUrl: "https://testpilot.firefox.com/"
};
```

Then, include that in the list of background scripts like so:
```json
  "background": {
    "scripts": [
      "currentEnvironment.js",
      "background.js"
    ]
  }
```
Any global vars you create in the first script should be available in the second, so you could leave lines 34-37 out of `background.js` altogether

Reply:
Oh yeah that's a way simpler way to do that, I'll fix it up in the morning. Thanks!